components/centraldashboard/public/components/main-page.js

```diff
@@ -27,88 +28,130 @@ import '@polymer/neon-animation/animations/fade-out-animation.js';
 import css from './main-page.css';
 import template from './main-page.pug';
 
+import './dashboard-view.js';
+import './activity-view.js';
+
 /**
  * Entry point for application UI.
  */
 export class MainPage extends PolymerElement {
     static get template() {
-        return html([`<style>${css.toString()}</style> ${template()}`]);
+        return html([`
+        <style is="custom-style"
+            include="iron-flex iron-flex-alignment iron-positioning">
+        <style>${css.toString()}</style> ${template()}
+        `]);
     }
 
     static get properties() {
         return {
-            links: {
+            page: String,
+            routeData: Object,
+            subRouteData: Object,
+            iframeRoute: Object,
+            menuLinks: {
                 type: Array,
                 value: [
-                    {text: 'Home', defaultPage: true, hasDivider: true},
                     {
-                        link: 'https://www.kubeflow.org/docs/about/kubeflow/',
+                        iframeUrl: 'https://www.kubeflow.org/docs/about/kubeflow/',
                         text: 'Kubeflow docs',
+                        href: '/docs',
```

Comment:
Let's call this `route` so it's clear this is a route for the current page rather than a _Hypertext **REFerence**_.

Reply:
See my above comment for my rationale. I'm not opposed to this change but LMK what you think.